brightwind/load/load.py

```diff
@@ -961,71 +1016,128 @@ def _get_id_token():
+                raise _BrighthubAuth.BrighthubAuthError(
+                    _BrighthubAuth.__BRIGHTHUB_LOGIN_ERROR_MAP["unexpected_error"])
 
         # login successful
         id_token = login_response['AuthenticationResult']['IdToken']
         refresh_token = login_response['AuthenticationResult']['RefreshToken']
+        return id_token, refresh_token
 
-        _BrighthubAuth.ID_TOKEN = id_token
-        _BrighthubAuth.REFRESH_TOKEN = refresh_token
+
+    @staticmethod
+    def _get_id_token():
+        """
+        Retrieve an ID token for authenticating BrightHub API requests.
+
+        This method attempts to authenticate using API client credentials if
+        `BRIGHTHUB_CLIENT_ID` and `BRIGHTHUB_CLIENT_SECRET` are available. If they
+        are not provided, it falls back to username/password authentication using
+        `BRIGHTHUB_EMAIL` and `BRIGHTHUB_PASSWORD`.
+
+        Username/password authentication is deprecated and will be removed in
+        version 3.0.0. Users should migrate to API key–based authentication.
+
+        Upon success, this method stores the resulting ID token (and refresh token,
+        when applicable) in the class-level attributes `ID_TOKEN` and
+        `REFRESH_TOKEN`. In case of authentication failure, an appropriate
+        exception is raised by the underlying authentication method.
+
+        :rtype: dict
+        :return: An empty dictionary upon successful authentication.
+        """
+        if _BrighthubAuth.REFRESH_TOKEN:
+            try:
+                _BrighthubAuth._brighthub_refresh_token()
+                return {}
+            except Exception:
+                # If refresh fails, proceed to re-authenticate
+                pass
+
+        # Preferred authentication flow: client credentials (API key)
+        try:
+            _BrighthubAuth.ID_TOKEN = _BrighthubAuth._authenticate_with_client_credentials()
+            return {}
+        
+        except Exception as e:
+            if str(e) in [
+                f'BRIGHTHUB_CLIENT_ID environmental variable is not set.',
+                f'BRIGHTHUB_CLIENT_SECRET environmental variable is not set.']:
+                # Fallback to basic auth if client credentials are not provided
+                # This method is to be depreciated in version 3.0.0
+                _BrighthubAuth.ID_TOKEN, _BrighthubAuth.REFRESH_TOKEN = _BrighthubAuth._authenticate_with_basic_auth()
+            else:
+                raise e
 
         return {}
+    
 
     @staticmethod
     def _brighthub_refresh_token():
         """
         Function to generate a new token if the current id_token has expired. The new tokens are assigned to the global
         variables ID_TOKEN, REFRESH_TOKEN.
         In case of an error, a error message will be returned
-
         """
-        url, headers, client_id = _BrighthubAuth._get_cognito_request()
+        warnings.warn(
```

Comment:
With the flow you have it seems like this _brighthub_refresh_token is called reagardless if the user is using API Keys. Therefore, this warning msg will appear no matter what method. It seems like this refresh token function is linked to the username and password method of authenticating because of this warning msg. Or maybe it is only called if the method was username and password as that is the only method that has a refresh token?

We probably only need one warning msg to show to the user if they are using the username and password method.